### PR TITLE
Add analytics opt in screen for newly registered users

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-meta/issues/238

This adds a new analytics opt-in screen for newly registered users. It aims at increasing the opt-in rate so that we can have better insights as to what journey of that category of users looks like

The analytics toast flow remains the same

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add analytics opt in screen for newly registered users ([\#8995](https://github.com/matrix-org/matrix-react-sdk/pull/8995)). Fixes vector-im/element-meta#238.

### CODE REVIEW

**Comment by weeman1337** [comment]
Do we already have a sign-up end-2-end test? I think we should have at least one clicking through the happy path.

File: src/components/structures/auth/AnalyticsOptIn.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import dis from "../../../dispatcher/dispatcher";
+import {
+    ButtonClicked,
+    showDialog as showAnalyticsLearnMoreDialog,
+} from '../../views/dialogs/AnalyticsLearnMoreDialog';
+import AccessibleButton from '../../views/elements/AccessibleButton';
+import SplashPage from '../SplashPage';
+import { Icon as AnalyticsIcon } from "../../../../res/img/element-icons/analytics.svg";
+
+interface IProps {
```
Comment (weeman1337) [inline src/components/structures/auth/AnalyticsOptIn.tsx:31]: Our new code style says no `I` any more

https://github.com/vector-im/element-meta/pull/216/files#diff-97f46bd1ed0e6dfce7a7002e96f72ecd446b90c5eb5f48d687dc2d9df856290cR127

**Comment by gsouquet** [comment]
@weeman1337 all done! This screen is not always shown as it depends on the react SDK config too. I've catered for that in my test

**Review by weeman1337 — COMMENTED** [review]
Tested on the Netlify deployment. Works :+1: 

If you want you can add some unit-tests for the new components :wink:

**Review by justjanne — APPROVED** [review]

**Pull request closed without merging**